### PR TITLE
task done

### PR DESCRIPTION
**Pull request by yutsevich**

### CODE REVIEW

**Review by andriylagun — APPROVED** [review]
Good job!

File: src/main/java/com/internet/shop/dao/ProductDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    public List<Product> getAll() {
+        return Storage.products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        List<Product> products = getAll();
+        IntStream.range(0, products.size())
+                .filter(i -> products.get(i).getId().equals(product.getId()))
+                .forEach(index -> products.set(index, product));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/ProductDaoImpl.java]:
```suggestion
                .forEach(i -> products.set(i, product));
```

Comment (dmytro-hryhoruk) [reply src/main/java/com/internet/shop/dao/ProductDaoImpl.java]: it's better not to change the name of the stream elements(only after map() operation)

File: src/main/java/com/internet/shop/servise/ProductServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package com.internet.shop.servise;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/servise/ProductServiceImpl.java]:
```suggestion
package com.internet.shop.servise.impl;
```
let's move implementations to separate sub-package

File: src/main/java/com/internet/shop/dao/ProductDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package com.internet.shop.dao;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/ProductDaoImpl.java]:
```suggestion
package com.internet.shop.dao.impl;
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**